File: docs/snippets/all/descriptors/descr_builtin_archetype.py
#!/usr/bin/env python3

from __future__ import annotations

import rerun as rr  # pip install rerun-sdk

rr.init("rerun_example_descriptors_builtin_archetype")
rr.spawn()

rr.log("data", rr.Points3D([[1, 2, 3]], radii=[0.3, 0.2, 0.1]), static=True)

# The tags are indirectly checked by the Rust version (have a look over there for more info).

File: docs/snippets/all/descriptors/descr_builtin_archetype.cpp
#include <rerun.hpp>

int main(int argc, char* argv[]) {
    const auto rec = rerun::RecordingStream("rerun_example_descriptors_builtin_archetype");
    rec.spawn().exit_on_failure();

    rec.log_static("data", rerun::Points3D({{1.0f, 2.0f, 3.0f}}).with_radii({0.3f, 0.2f, 0.1f}));

    // The tags are indirectly checked by the Rust version (have a look over there for more info).
}
